feat: add Root.io unaffected vulnerability  filtering

chait-slim opened the pull request:
- Add common helper functions to identify Root.io patched packages
  - Implement filtering logic to remove false positives for .root.io versions
  - Integrate filtering into APK, DPKG, and Python matchers
  - Add comprehensive unit tests for filtering logic

  Packages with .root.io in their version string are patched by Root.io
  and should not report vulnerabilities marked as ROOTIO_UNAFFECTED.

  Signed-off-by: Chai Tadmor <chai.tadmor@root.io>

willmurphyscode commented:
Hi @chait-slim! Thanks for addressing the feedback.

A couple things: 

First: I think we'd benefit from a call to talk about this new vulnerability data and how it should work with Grype - I am trying to give sort of high level feedback in GitHub comments and I think some context is being lost. Can you make an appointment with me at https://calendar.app.google/vtAP61H34Cu4QQWq7 ? Please include anyone who can answer questions about the semantics of the vulnerability data. Thanks!

Second: the PRs are failing the DCO check. See https://github.com/anchore/grype/blob/main/CONTRIBUTING.md#sign-off-your-work for some info on how to fix this and why it is required.

chait-slim commented:
> Hi @chait-slim! Thanks for addressing the feedback.
> 
> A couple things:
> 
> First: I think we'd benefit from a call to talk about this new vulnerability data and how it should work with Grype - I am trying to give sort of high level feedback in GitHub comments and I think some context is being lost. Can you make an appointment with me at https://calendar.app.google/vtAP61H34Cu4QQWq7 ? Please include anyone who can answer questions about the semantics of the vulnerability data. Thanks!
> 
> Second: the PRs are failing the DCO check. See https://github.com/anchore/grype/blob/main/CONTRIBUTING.md#sign-off-your-work for some info on how to fix this and why it is required.

Thanks @willmurphyscode!
Ive booked some time to discuss this and the other comment as well

chait-slim commented:
Opened a new PR with updated spec: https://github.com/anchore/grype/pull/3137
